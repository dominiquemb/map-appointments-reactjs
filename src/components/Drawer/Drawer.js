import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import { Drawer as MuiDrawer, Hidden, ListItem } from '@material-ui/core';

// import logo from '../../assets/images/logo.png';


const styles = theme => ({
  logo: {
    display: 'block',
    margin: '0 auto',
  },
  logoContainer: {
    backgroundColor: theme.palette.secondary.main,
    marginBottom: 20,
    marginTop: 15,
  },
  drawerPaper: {
    borderRight: 0,
    width: theme.drawer.width,
    [theme.breakpoints.up('md')]: {
      position: 'fixed',
    },
    backgroundColor: theme.palette.secondary.main,
  },
});

const propTypes = {
  classes: PropTypes.oneOfType([PropTypes.object]).isRequired,
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

const Drawer = ({
  classes,
  children,
}) => (
  <Hidden mdDown>
    <MuiDrawer variant="permanent" open classes={{ paper: classes.drawerPaper }}>
      <ListItem className={classes.logoContainer} disableGutters>
        {/* <img src={logo} width="120px" className={classes.logo} alt="Logo" /> */}
      </ListItem>
      {children}
    </MuiDrawer>
  </Hidden>
);

Drawer.propTypes = propTypes;
Drawer.defaultProps = defaultProps;
const DrawerStyled = withStyles(styles, { withTheme: true })(Drawer);
export default withRouter(DrawerStyled);
