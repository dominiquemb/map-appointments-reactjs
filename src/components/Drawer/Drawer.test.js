import React from 'react';
import { shallow } from 'enzyme';
import Drawer from './Drawer';

const props = {
  mobileOpen: false,
  role: 'ROLE_PHARMACIST',
  toggleHandler: () => {},
};

describe('Testing Drawer', () => {
  const wrapper = shallow(<Drawer {...props} />);
  test('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
