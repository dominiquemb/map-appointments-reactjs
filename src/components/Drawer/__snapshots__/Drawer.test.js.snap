// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Drawer renders as expected 1`] = `ShallowWrapper {}`;
